test/verify/check-kdump

```diff
@@ -486,5 +514,162 @@ class TestKdumpNFS(KdumpHelpers):
+        b_cockpit.set_val("#kdump-settings-location", "local")
+        pathInput = "#kdump-settings-local-directory"
+        b_cockpit.set_input_text(pathInput, customPath)
+        b_cockpit.click(f"#kdump-settings-dialog button{self.primary_btn_class}")
+        b_cockpit.wait_text("#kdump-change-target", f"locally in {customPath}")
+
+        # Create a fake repo so dnf does not hang trying to refresh the metadata cache
+        # which requires internet access.
+        self.createPackage("kexec-tools", "1", "1", content={"/kexec-tools": ""})
+        self.enableRepo()
+
+        # HACK: workaround ansible role not creating /var/crash2 https://github.com/linux-system-roles/kdump/issues/8
+        m.execute(f"mkdir -p {customPath}")
+
+        # This requires a reboot
+        self.run_ansible_playbook(b_cockpit, allow_failure=True)
+        m.reboot()
+
+        self.login_and_go("/kdump")
+        m.execute("until systemctl is-active kdump; do sleep 1; done")
+
+        # Verify that kdump runs and crashkernel is configured
+        b.wait_in_text("#app", "Service is running")
+        b.wait_in_text("#app", "192 MiB")
+        self.assertActive(active=True)
+
+        # Second run should be succeed as crashkernel is now set in the kernel cmdline
+        self.run_ansible_playbook(b)
+
+        conf = m.execute("cat /etc/kdump.conf")
+        self.assertIn("\n# Ansible managed\n", conf)
+        self.assertIn("\npath /var/crash2\n", conf)
+
+        self.crashKernel("stored in /var/crash2 as vmcore")
+        m.execute(f"until test -e {customPath}/127.0.0.1*/vmcore; do sleep 1; done", timeout=180)
+        self.assertIn("Kdump compressed dump", m.execute(f"file {customPath}/127.0.0.1*/vmcore"))
+
+        # Test SSH
+        b_cockpit.click("#kdump-change-target")
+        b_cockpit.set_val("#kdump-settings-location", "ssh")
+        sshInput = "#kdump-settings-ssh-server"
+        b_cockpit.set_input_text(sshInput, "root@localhost")
+        sshKeyInput = "#kdump-settings-ssh-key"
+        pathInput = "#kdump-settings-ssh-directory"
+        b_cockpit.set_input_text(sshKeyInput, "/root/.ssh/id_rsa")
+        b_cockpit.set_input_text(pathInput, "/var/crash")
+        b_cockpit.click(f"#kdump-settings-dialog button{self.primary_btn_class}")
+        b_cockpit.wait_not_present(pathInput)
+        b_cockpit.wait_text("#kdump-change-target", "Remote over SSH")
+
+        self.enableLocalSsh()
+        self.run_ansible_playbook(b_cockpit)
+        conf = m.execute("cat /etc/kdump.conf")
+        self.assertIn("\n# Ansible managed\n", conf)
+        self.assertIn("\nssh root@localhost\n", conf)
+        self.assertIn("\nsshkey /root/.ssh/id_rsa\n", conf)
+        self.assertIn("\npath /var/crash\n", conf)
+
+        self.login_and_go("/kdump")
+        b.wait_text("#kdump-change-target", "Remote over SSH")
+        self.crashKernel("copied through SSH to root@localhost:/var/crash as vmcore", cancel=True)
```

Comment:
Please add a test for an SSH target without user name. This doesn't have to go through the full crash/reboot cycle, just checking the generated kdump.conf.

Reply:
This is still current -- confirmed by coverage report.

Reply:
Right, but then we won't test whatever ansible generates hmm wonder if I can force it to not run kdumpctl, we also need to report that brokenness it might be legit. (But I kinda grew tired of this all).

Reply:
How do you mean? Running the ansible playbook should *never* reboot or crash the system, these are actions that the user does on the cockpit page. I meant configure kdump like that (ssh target without user), run ansible, and check the generated kdump.conf. No need to reboot/crash afterwards.

Reply:
(If you are concerned about that rebuilding the initrd, then so be it..)

Reply:
Running the ansible role again on the same machine breaks `kdumpctl` because of pty for some annoying reason. That has nothing to do with rebooting or crashing just checking if the ansible role creates the correct /etc/kdump.conf. It is incredibly annoying.

Reply:
Correction, it breaks `kdump.service` :exploding_head: 

```
Nov 23 11:33:40 fedora-39-127-0-0-2-2201 systemd[1]: Starting kdump.service - Crash recovery kernel arming...
Nov 23 11:33:40 fedora-39-127-0-0-2-2201 kdumpctl[6512]: Pseudo-terminal will not be allocated because stdin is not a terminal.
Nov 23 11:33:40 fedora-39-127-0-0-2-2201 kdumpctl[6506]: kdump: Starting kdump: [FAILED]
```

No idea how one debugs this?

Reply:
That's the diff between the previous kdump.conf which worked, and the one which causes `systemctl start kdump` to fail:

```diff
--- /etc/kdump.conf.withuser	2023-11-23 12:42:47.770730648 +0000
+++ /etc/kdump.conf	2023-11-23 12:45:10.958730648 +0000
@@ -4,11 +4,11 @@
 # system_role:kdump
 
 
-ssh admin@localhost
+ssh localhost
 
 sshkey /root/.ssh/id_rsa
 
-path /var/crash
+path /var/crash2
 core_collector makedumpfile -l --message-level 7 -d 31 -F
 
 auto_reset_crashkernel yes
```

Replacing the conf with the .withuser one succeeds, so it's not really about "second time", but that the config is broken somehow, and kdumpctl doesn't tell you. The `path` change is irrelevant, it also fails with `/var/crash`. But changing the `ssh` line from `localhost` to `root@localhost` makes it go kaboom :boom: .

So this cries for a kdump bugzilla. This is pretty much impossible to work around -- we *really* don't want to assume `root@`, as /root/.ssh/config plausibly has a config stanza for that remote host, and we shouldn't overwite that. Maybe the role does that wrong, or it documents that the user will explicitly be specified as `root` when not given, but apparently it doesn't actually do that.

Reply:
I filed https://bugzilla.redhat.com/show_bug.cgi?id=2251216 . But looking at the manpage, it is actually specified as

> ssh <user@server>

So I think this is a bug in the kdump role.

Reply:
Summary from meeting: Let's stop writing `.location`. That removes the redundancy, and sidesteps that bug. If that works, let's adjust https://github.com/linux-system-roles/kdump/issues/184 to update the documentation/examples, or whatever else led you to setting `.location` in the first place.

Reply:
Ack, this requires us to parse the server value and get the username and server out. Otherwise we'll get admin@admin@server.

as server would contain `admin@server` and the user would be `admin`.

Reply:
Ack, this requires us to parse the server value and get the username and server out. Otherwise we'll get admin@admin@server.

as server would contain `admin@server` and the user would be `admin`. I've commented on the kdump ansible role issue as well

Reply:
Right, but it already does that split and uses the left half of it, right? so it could just as well use the right side too.